src/api/apiRoutes.js

```diff
@@ -0,0 +1,47 @@
+import express from "express"
+import { getGames } from "./helpers.js"
+
+const router = express.Router()
+
+router.get("/get-top-products", (req, res) => {
+  const games = getGames()
+  const gamesSortedByDate = games
+    .sort((a, b) => a.releaseDate - b.releaseDate)
+    .slice(0, 3)
+  return res.send(gamesSortedByDate)
+})
+
+// GAME PRODUCTS LIST
+router.get(`/products`, (req, res) => {
+  const listOfGames = getGames()
+  let searchQuery = req.query.search?.toLowerCase()
+  const categoryQuery = req.query.category
+  const filteredByCategory = listOfGames.filter((game) =>
+    game.category.includes(categoryQuery)
+  )
+  if (!searchQuery && !categoryQuery) {
+    return res.send(listOfGames)
+  }
+  if (searchQuery === undefined) {
+    searchQuery = ""
+  }
+  if (searchQuery && !categoryQuery) {
```

Comment:
Why do you need several almost simalar filters and if-statements if you could use one single array of games and re-filter it when you need it.

Example
`
 let games = ...
 if (searchQuery) {
   games = games.filter(...)
 }
 
 if (categoryQuery) {
   games = games.filter(...)
 }
 
 res.send(games)
`